spring-integration-core/src/main/java/org/springframework/integration/endpoint/PollingConsumer.java

```diff
@@ -72,6 +75,26 @@ public void setReceiveTimeout(long receiveTimeout) {
 	}
 
 	@Override
+	public String getInputChannelName() {
+		return this.inputChannel == null ? null : this.inputChannel instanceof NamedComponent ?
+			((NamedComponent) this.inputChannel).getComponentName() : null;
+	}
+
+
+	@Override
+	public String getOutputChannelName() {
+		MessageChannel channel = this.handler instanceof AbstractMessageProducingHandler ?
+				((AbstractMessageProducingHandler) this.handler).getOutputChannel() : null;
```

Comment:
Well, after moving `getOutputChannelName()` to the `MessageProducer`, we don't need to worry about proxy here any more! 
Plus that `getOutputChannel()`  might be redundant...